feat(rule-engine): add bank account verification operator

tomer-shvadron opened the pull request:
- Implement BankAccountVerification class for rule evaluation
- Update relevant enums and constants for integration
- Add unit tests for verifying rules based on bank account data


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Introduced support for bank account verification rules, enabling evaluation of specific rule scores for account verification scenarios.

- **Tests**
  - Added comprehensive tests to verify correct behavior of bank account verification rule evaluation.

- **Documentation**
  - Updated schemas to include the new bank account verification operator for improved clarity in rule definitions.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

tomer-shvadron committed:
feat(rule-engine): add bank account verification operator

- Implement BankAccountVerification class for rule evaluation
- Update relevant enums and constants for integration
- Add unit tests for verifying rules based on bank account data

tomer-shvadron commented:
Reminder: Update packages before merging

alonp99 commented on packages/common/src/rule-engine/operators/bank-account-verification.ts:
```diff
@@ -0,0 +1,98 @@
+      throw new ValidationFailedError('Extract value', 'parsing failed', result.error);
+    }
+
+    if (result.data.requestType !== 'BAVCommercial-Standard') {
```
It looks like its being reused so i would extract it to a const

tomer-shvadron replied:
I didn't understand, what is reused? `requestType` serves no purpose before or after this line, same goes for the string

alonp99 replied:
the string, in tests

alonp99 commented on packages/common/src/rule-engine/operators/bank-account-verification.ts:
```diff
@@ -0,0 +1,98 @@
+    }
+
+    const rules = rulesResult.data.filter(rule =>
+      ['CMM1069', 'CMM1048', 'CMM1052'].includes(rule.ruleId),
```
maybe also extract it to consts and leave a comment about what they say?

tomer-shvadron committed:
refactor(bank-account-verification): simplify rule filtering logic

- Remove unnecessary inline rule array and replace with a constant
- Add comment for clarity on required rules for bank account verification

tomer-shvadron committed:
feat(rule-engine): add bank account verification request type constant

- Introduce BANK_ACCOUNT_VERIFICATION_COMMERCIAL_REQUEST_TYPE constant
- Refactor code to use constant in validation checks
- Update unit tests with new constant for consistency

tomer-shvadron committed:
chore(deps): update dependencies across multiple packages

- Bump version of common package to 0.9.108
- Update workflow-related SDKs to version 0.6.131
- Increment versions for several applications and services
